pkg/minikube/mount/cifs_windows.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh/terminal"
+	"k8s.io/minikube/pkg/minikube/drivers/hyperv"
+	"k8s.io/minikube/pkg/minikube/out"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var credentialName = "minikube"
+
+func (m *Cifs) Share() error {
+	// Ensure that the current user is administrator because creating a SMB Share requires Administrator privileges.
+	err := hyperv.IsWindowsAdministrator()
+	if err != nil {
+		return errors.Wrap(err,"current user not administrator")
+	}
+
+	// Check if Name of the Share already exists or not.
+	err = hyperv.PowershellCmd("SmbShare\\Get-SmbShare","-Name",m.HostShareName)
+	if err == nil {
+		glog.Infof("The share with share name %v already exists. Trying to delete it.", m.HostShareName)
+		if err := hyperv.PowershellCmd("SmbShare\\Remove-SmbShare", "-Name", m.HostShareName, "-Force"); err != nil {
+			return errors.Wrap(err, "remove smb share")
+		}
+		glog.Infof("The share with share name %v has been deleted", m.HostShareName)
+	}
+
+	// Get the current user so that we can assign full access permissions to only that user.
+	user, err := hyperv.CurrentWindowsUser()
+	if err != nil {
+		return errors.Wrap(err, "get current user")
+	}
+	glog.Infof("Current User -- [%v]",user)
+	glog.Infof("Trying to enable share for CIFS Mounting.")
+	if err := hyperv.PowershellCmd("SmbShare\\New-SmbShare", "-Name", m.HostShareName, "-Path", m.HostPath , "-FullAccess", user, "-Temporary"); err != nil {
+		return errors.Wrap(err, "new smb share")
+	}
+	return nil
+}
+
+func (m *Cifs) Unshare() error {
+
+	// Ensure that the current user is administrator because creating a SMB Share requires Administrator privileges.
+	err := hyperv.IsWindowsAdministrator()
+	if err != nil {
+		return errors.Wrap(err,"current user not administrator")
+	}
+
+	if err := hyperv.PowershellCmd("SmbShare\\Remove-SmbShare", "-Name", m.HostPath, "-Force"); err != nil {
+		return errors.Wrap(err,"remove smb share")
+	}
+	return nil
+}
+
+func (m *Cifs) Mount(r mountRunner) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return errors.Wrap(err, "get hostname")
+	}
+	user, err := hyperv.CurrentWindowsUser()
+	if err != nil {
+		return errors.Wrap(err,"get current user")
+	}
+
+	// We need the Domain Name of the current user and the User Name. This is used while connecting to the Share from Linux.
+	stringSplitResult := strings.Split(user,"\\")
+	domain := stringSplitResult[0]
+	username := stringSplitResult[1]
+
+	// Only ask the user for the password if it is not found in the Windows Credential Store
+	mountPassword,err := getMountPassword()
+	if mountPassword == "" || err != nil  {
+		err = saveMountPassword()
+		if err != nil {
+			return errors.Wrap(err, "save password to wincred")
+		}
+	}
+
+	mountCmd := fmt.Sprintf("sudo mkdir -p %s && sudo mount.cifs //%s/%s %s -o username=%s,password=%s,domain=%s",m.VMDestinationPath, hostname, m.HostShareName, m.VMDestinationPath, username, mountPassword , domain)
```

Comment:
Use `credentials` instead of `password`, to avoid leaking the password onto the command-line.

I'd also recommend deleting the credentials file once mount succeeds.